File: 790. Domino and Tromino Tiling.java
//790. Domino and Tromino Tiling


class Solution {
    public int numTilings(int n) {
        long a = 1;
        long b = 2;
        long c = 5;
        long mod = 1000000007;
        if(n == 1){
            return (int)a;
        }
        if(n == 2){
            return (int)b;
        }
        if(n == 0){
            return 0; 
        }
        for(int i = 4 ; i <= n ;i++){
            long k = ((c*2) + a)%mod;
            a = b;
            b = c;
            c = k;            
        }
        return (int)c;
    }
}

File: 790. Domino and Tromino Tiling.js
//790. Domino and Tromino Tiling


/**
 * @param {number} n
 * @return {number}
 */
var numTilings = function(n) {
    let a = 1;
    let b = 2;
    let c = 5;
    let mod = 1000000007;
    if(n == 1){
        return a;
    }
    if(n == 2){
        return b;
    }
    if(n == 0){
        return 0; 
    }
    for(let i = 4 ; i <= n ;i++){
        let k = ((c*2) + a)%mod;
        a = b;
        b = c;
        c = k;            
    }
    return c;

};
